### DISCUSSION TITLE
NoDecodeDelegateForThisImageFormat

### DISCUSSION THREAD

**Discussion by asze17** [ImageMagick/ImageMagick#3710] [Help] [answered]
I am using the python binding: [Wand](https://github.com/emcconville/wand)
Every time I tried to open an image it raise this error:
```
wand.exceptions.MissingDelegateError: NoDecodeDelegateForThisImageFormat `PNG' @ error/blob.c/BlobToImage/458
```
I am really unsure why because like it used to work perfectly fine then suddenly it just raise that error and I did nothing to code nor ImageMagick or Wand.

**Comment by fmw42** [comment]
Please show your code!  Looks like libpng is missing or ImageMagick is compiled without PNG delegate or perhaps no delegates.

**Reply by asze17** [reply]
> Please show your code! Looks like libpng is missing or ImageMagick is compiled without PNG delegate or perhaps no delegates.

```py
from wand.image import Image
from io import BytesIO
with Image(file=BytesIO(await author.avatar_url_as(format="png").read()), format="png") as img:
 ...
 ```
That was the code I used for testing. And there could be a chance that ImageMagick is compiled without PNG delegate but I am unsure how to check that.

**Comment by snibgo** [comment]
> And there could be a chance that ImageMagick is compiled without PNG delegate but I am unsure how to check that.

If you have IM v7, what is the text output from "magick -version"?

If you have IM v6, what is the text output from "convert -version"?

**Reply by asze17** [reply]
> > And there could be a chance that ImageMagick is compiled without PNG delegate but I am unsure how to check that.
> 
> If you have IM v7, what is the text output from "magick -version"?
> 
> If you have IM v6, what is the text output from "convert -version"?

Both of them returned `command not found`
I installed MagickWand with apt by `apt install libmagickwand-dev` as instructed in [wand's installation guide](https://docs.wand-py.org/en/0.6.6/guide/install.html#install-imagemagick-on-debian-ubuntu)

**Comment by asze17** [comment, marked as answer]
I am closing this as it just fixed itself magicklly I have no idea what happened but it just managed to fix itself
